Stratis.Bitcoin/Features/BlockStore/BlockStoreCachePerformanceCounter.cs

```diff
@@ -12,12 +12,22 @@ public class BlockStoreCachePerformanceCounter
 		private long _CacheRemoveCount;
 		private long _CacheHitCount;
 		private long _CacheMissCount;
+        private string _Name;
 
-		public BlockStoreCachePerformanceCounter()
+		public BlockStoreCachePerformanceCounter(string name = "BlockStore")
 		{
+            this._Name = name;
 			this._Start = DateTime.UtcNow;
 		}
 
+        public string Name
```

Comment:
Personally I would like to see "get; private set;" in the interest of brevity but unfortunately "Interlocked" requires a variable and not a property. This means we need separate variables and separate getters.